### PR TITLE
Migrating to wyhash v3 #115

### PR DESCRIPTION
**Pull request by ivte-ms**
Migrating to wyhash v3, specifically to WYHASH32 version of it to have consistent perf. between Hololens 1 and 64-bit platforms.

The code is mostly copy-pasted from the current version of https://github.com/wangyi-fudan/wyhash/blob/master/wyhash.h, which is licensed under Unlicense (public domain). The changes from the original include:
- applying clang-format (to avoid special-casing this file) and splitting lines that declare multiple variables into individual lines.
- Adding noexcept/constexpr/MS_MR_SHARING_FORCEINLINE in a few places.
- Eliminating warnings related to implicit narrowing casts (by introducing static_cast there).
- Extracting MS_MR_LIKELY/MS_MR_UNLIKELY (it will be useful in a few other places).
- Making sure that size arg stays size_t (it's uint64_t in the original source).
- Removing parts related to the random number generator (unused; plus it would be thread-unsafe anyway).

The output of the hash is different from previously used v2, but should be identical to the reference implementation of v3 with WYHASH32 define.

Bug: #115

### CODE REVIEW

**Comment by ivte-ms** [comment]
The latest revision also fixes a bug in the reference implementation of wyhash, see https://github.com/wangyi-fudan/wyhash/issues/36

(if some better mitigation will be discovered, I'll update our version as well).

File: libs/Common-cpp/src/hash.cpp [hunk 1]
```diff
@@ -87,8 +87,10 @@ MS_MR_SHARING_FORCEINLINE uint64_t _wyr3(const uint8_t* p, unsigned k) {
 uint64_t CalculateHash64(const char* data,
                          size_t size,
                          uint64_t seed) noexcept {
+  // This is a deviation from the current reference implementation of wyhash
+  // (it would always return 0).
   if (MS_MR_UNLIKELY(!size))
-    return 0;
+    return _wymum(_wymum(seed ^ _wyp0, _wyp1), _wyp2);
```
Comment (djee-ms) [inline libs/Common-cpp/src/hash.cpp]: Did we report that back to the original author? Maybe that's a worthy contribution, no?

Comment (wangyi-fudan) [reply libs/Common-cpp/src/hash.cpp]: got it. I use：
_wymum(_wymum(seed^_wyp0,seed^_wyp1),_wyp4); 
for the consisitancy with rest of the code.

Comment (ivte-ms) [reply libs/Common-cpp/src/hash.cpp]: Updated the code to match the reference implementation.

**Pull request merged**